api/addons/serializers.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import serializers as ser
+from api.base.serializers import JSONAPISerializer, LinksField
+from api.base.utils import absolute_reverse
+
+class NodeAddonFolderSerializer(JSONAPISerializer):
+    """
+    Overrides AddonSettingsSerializer to return node-specific fields
+    """
+    class Meta:
+        type_ = 'node_addon_folders'
+
+    id = ser.CharField(source='provider', read_only=True)
+    kind = ser.CharField(default='folder', read_only=True)
+    name = ser.CharField(read_only=True)
+    path = ser.CharField(read_only=True)
+    folder_id = ser.CharField(read_only=True)
+
+    links = LinksField({
+        'children': 'get_child_folders'
+    })
+
+    def get_child_folders(self, obj):
```

Comment:
This serializes folder data returned from the third party API of any given addon. As noted in this [`TODO`](https://github.com/CenterForOpenScience/osf.io/pull/5532/files#diff-51eef6ea2a6c35e2b080c9a6fe591e38R1487), the `<Addon>NodeSettings` models (and likely tests, generic views) require a moderate refactor in order to make this generalizable via APIv2. Given that the only existing use case we have for this feature is for GDrive, and that it was requested a while ago, and that it will need post-model-refactoring refactoring, I wrote this part of the feature somewhat hastily.